File: leetcode/2016. Maximum Difference Between Increasing Elements/2016_maximum_difference_between_increasing_elemets.cs
public class Solution {
    public int MaximumDifference(int[] nums) {
        int output = -1;
        int minValue = nums[0];
        for (int i=1; i<nums.Length; i++)
        {
            if(nums[i] > minValue)
            {
                output = Math.Max(output, nums[i]-minValue);
            }
            else
            {
                minValue = Math.Min(minValue, nums[i]);
            }
        }
        return output;        
    }
}

File: leetcode/2016. Maximum Difference Between Increasing Elements/2016_maximum_difference_between_increasing_elemets.py
class Solution:
    def maximumDifference(self, nums: List[int]) -> int:
        output = 0
        for i in range(len(nums)):
            for j in range(i+1, len(nums)):
                if nums[i] < nums[j]:
                    output = max(nums[j]- nums[i], output)
        
        return -1 if output == 0 else output
